joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -724,6 +738,57 @@ bool JointTrajectoryController::validate_trajectory_msg(
       return false;
     }
   }
+
+  rclcpp::Duration previous_traj_time(0);
+  for (auto i = 0ul; i < trajectory.points.size(); ++i) {
+    if ((i > 0) && (rclcpp::Duration(trajectory.points[i].time_from_start) <= previous_traj_time)) {
+      RCLCPP_ERROR(
+        lifecycle_node_->get_logger(),
+        "Time between points %u and %u is not strictly increasing, it is %f and %f respectively",
+        i - 1, i, previous_traj_time.seconds(),
+        rclcpp::Duration(trajectory.points[i].time_from_start).seconds());
+      return false;
+    }
+    previous_traj_time = trajectory.points[i].time_from_start;
+
+    if (trajectory.joint_names.size() != trajectory.points[i].positions.size()) {
```

Comment:
Nitpick: `trajectory.joint_names.size()` is written 8 times in this section. Perhaps a slightly shorter alternative could be created.

Also, I think a clever function could remove the repetitive parts of the "check this vector is empty then check for size, then get_logger(), general bla % parameters". 

```
bool validate_trajectory_point_vector_field(joint_names_size, vector_field, string_for_vector_field, i)
{
    if (!vector_field.empty() &&
      joint_names_size !=vector_field.size())
    {
      RCLCPP_ERROR(
        lifecycle_node_->get_logger(),
        "Mismatch between joint_names (%u) and %s (%u) at point #%u.",
        joint_names_size, string_for_vector_field, vector_field.size(), i);
      return false;
    }
}
```

and `&&` the results together and check whether it's false to early exit this loop?

Reply:
Done